File: python/P118.py
class Solution(object):
    def generate(self, numRows):
        if numRows == 0:
            return []
        elif numRows == 1:
            return [[1]]
        elif numRows == 2:
            return [[1],[1,1]]
        else:
            answer = [[1], [1,1]]
            for i in range(2, numRows):
                length = i+1
                temp = [1]*length
                for j in range(1, length-1):
                    temp[j] = answer[i-1][j-1] + answer[i-1][j] 
                temp[length-1] = 1
                answer.append(temp)
            return answer

File: java/P118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> answer = new ArrayList<List<Integer>>();
        if (numRows <= 0) return answer;
        for (int i = 0; i < numRows; i++) {
            List<Integer> temp = new ArrayList<Integer>();
            for (int j = 0; j < i+1; j++) {
                if (j == 0 || j == i) {
                    temp.add(1);
                } else {
                    temp.add(answer.get(i-1).get(j-1)+answer.get(i-1).get(j));
                }
            }
            answer.add(temp);
        }
        return answer;
    }
}